evm_loader/lib/src/types/tracer_ch_db.rs

```diff
@@ -229,18 +244,19 @@ impl ClickHouseDb {
 
     async fn get_account_rooted_slot(&self, key: &str, slot: u64) -> ChResult<Option<u64>> {
         info!("get_account_rooted_slot {{ key: {key}, slot: {slot} }}");
+
         let query = r#"
-            SELECT DISTINCT slot
-            FROM events.update_account_distributed
-            WHERE pubkey = ?
-                AND slot <= ?
-                AND slot IN (
-                    SELECT slot
-                    FROM events.update_slot
-                    WHERE status = 'Rooted'
-                )
-            ORDER BY slot DESC
-            LIMIT 1
+        SELECT DISTINCT uad.slot
+        FROM events.update_account_distributed AS uad
+        WHERE uad.pubkey = ?
+          AND uad.slot <= ?
+          AND (
+            SELECT COUNT(slot)
```

Comment:
This subquery isn't connected with the outer one. Can we extract this subquery to separated one and do the second query only on success?